transmart-core-db/grails-app/services/org/transmartproject/db/pedigree/RelationService.groovy

```diff
@@ -0,0 +1,27 @@
+package org.transmartproject.db.pedigree
+
+import grails.transaction.Transactional
+import groovy.transform.CompileStatic
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.pedigree.RelationResource
+import org.transmartproject.core.users.User
+import org.transmartproject.db.ontology.MDStudiesService
+
+@Transactional(readOnly = true)
+@CompileStatic
+class RelationService implements RelationResource {
+
+    @Autowired
+    MDStudiesService studiesService
+
+    @Override
+    List<org.transmartproject.core.pedigree.Relation> getAll(User user) throws AccessDeniedException {
+        if (studiesService.areAllStudiesPublic() || user.admin) {
```

Comment:
Just as a general remark:
There are generally two approaches in this kind of functions:
```java
if (errorCondition) {
  throw Exception()
}
doSomething()
```
or
```java
if (!errorCondition) {
  doSomething()
} else {
  throw Exception()
}
```

I personally like the first one more, because the error condition and the error response are together, and the indenting signals that the main behaviour (`doSomething()`) is part of the main flow, and not just some branch of behaviour. Do you agree?
It's a matter of code style, so I don't think there is only one good approach.

Reply:
yes, indeed the first approach looks better. In this case I used the second one, because this condition: `if(c1 || c2)` seemed simpler to me than `if(!c1 && !c2)`, but maybe better to be consistent and use the first one in all the cases. I updated the code here.